Src/AmrCore/AMReX_ErrorList.H

```diff
@@ -372,6 +376,86 @@ private:
 
 std::ostream& operator << (std::ostream& os, const ErrorList& elst);
 
+  struct AMRErrorTagInfo
+  {
+    int            m_max_level;
+    amrex::Real    m_min_time;
+    amrex::Real    m_max_time;
+    amrex::RealBox m_realbox;
+
+    AMRErrorTagInfo () noexcept : m_max_level(-1), m_min_time(-1), m_max_time(-1), m_realbox(amrex::RealBox()) {}
+
+    AMRErrorTagInfo& SetMaxLevel (int max_level) noexcept {
+      m_max_level = max_level;
+      return *this;
+    }
+    AMRErrorTagInfo& SetMinTime (amrex::Real min_time) noexcept {
+      m_min_time = min_time;
+      return *this;
+    }
+    AMRErrorTagInfo& SetMaxTime (amrex::Real max_time) noexcept {
+      m_max_time = max_time;
+      return *this;
+    }
+    AMRErrorTagInfo& SetRealBox (const amrex::RealBox& realbox) noexcept {
+      m_realbox = realbox;
+      return *this;
+    }
+  };
+
+  class AMRErrorTag
+  {
+  public:
+
+    enum TEST {GRAD=0, LESS, GREATER, VORT, BOX, USER};
+
+    struct UserFunc
+    {
+      virtual void operator() (const amrex::Box&                       bx,
```

Comment:
Is it simpler for the user to use a function pointer instead of a class derive from this?

Reply:
I would really like to avoid that path if possible, I'm not a fan of passing around function pointers if there's a more C++ way to do it.

Reply:
The interface I created had in mind several use cases I've had over the past, where a class was able to hold state info that was useful later.  The overhead of doing this is minimal in C++ and the flexibility it buys I think is worth it.

Reply:
That's true.